datalad_container/extractors/_load_singularity_versions.py

```diff
@@ -0,0 +1,35 @@
+"""
+Importing this file extends datalad.support.external_version:
+
+Adds:
+    - external_versions["cmd:apptainer"]
+    - external_versions["cmd:singularity"]
+"""
+import subprocess
+
+from datalad.cmd import (
+    StdOutCapture,
+    WitlessRunner,
+)
+from datalad.support.external_versions import external_versions
+
+
+def __get_apptainer_version():
+    version = WitlessRunner().run("apptainer --version", protocol=StdOutCapture)['stdout'].strip()
+    strip_str = len("apptainer version ")
+    return version[strip_str:]
+
+
+def __get_singularity_version():
+    version = WitlessRunner().run("singularity version", protocol=StdOutCapture)['stdout'].strip()
+
+    # It may be possible to have both apptainer and singularity installed.
+    # If singularity is installed independently, the versions will not match.
```

Comment:
is that guaranteed -- they really cannot be the same?

Reply:
Now that you mention it, I think it's ok to have cmd: singularity return the apptainer version.

It wouldn't be ideal as the code currently exists because it collects both singularity and apptainer version info, but once I use the dynamic command as you suggested, it's a moot point.

+1 I'm convinced